src/Contexts/Authentication/AuthenticationContext.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {useState, useContext, createContext, ReactNode} from "react";
+import SubmarineAuthenticationHttpClient from "../../SubmarineHttp/Resources/Authentication/SubmarineAuthenticationHttpClient";
+import SubmarineHttpVersion from "../../SubmarineHttp/SubmarineHttpVersion";
+import ISubmarineRegisterRequest from "../../SubmarineHttp/Resources/Authentication/Register/ISubmarineRegisterRequest";
+import ISubmarineAuthenticateRequest from "../../SubmarineHttp/Resources/Authentication/Authenticate/ISubmarineAuthenticateRequest";
+import ISubmarineAuthenticatedResponse from "../../SubmarineHttp/Resources/Authentication/Authenticate/ISubmarineAuthenticatedResponse";
+import AuthenticationIdentity from "./AuthenticationIdentity";
+import IAuthenticationContextProviderProps from "./IAuthenticationContextProviderProps";
+import IAuthenticationContext from "./IAuthenticationContext";
+
+const InitialAuthenticationContext = { register: () => {}, authenticate: () => { } } as IAuthenticationContext;
+const AuthenticationContext = createContext(InitialAuthenticationContext)
+
+export const AuthenticationContextProvider = ({ children }: IAuthenticationContextProviderProps) => {
+    const client = new SubmarineAuthenticationHttpClient(SubmarineHttpVersion.one);
+
+    const initialIdentity = AuthenticationIdentity.fromNoBearerToken();
+    const [identity, setIdentity] = useState(initialIdentity);
```

Comment:
Consider backloading this off to the session or even local storage.

Reply:
Perhaps just store the bearer token in storage and derive the identity from it, because identity is our wrapper, not the code itself.